bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java

```diff
@@ -1081,13 +1084,18 @@ public void setLength(long length) {
     }
 
     private void checkFixedLength(long length) {
-        if (arrayType != null && arrayType.getTag() == TypeTags.TUPLE_TAG) {
-            throw BLangExceptionHelper.getRuntimeException(BallerinaErrorReasons.INHERENT_TYPE_VIOLATION_ERROR,
-                    RuntimeErrors.CANNOT_CHANGE_TUPLE_SIZE);
-        }
-        if (((BArrayType) this.arrayType).getState() == ArrayState.CLOSED_SEALED) {
+        if (arrayType.getTag() == TypeTags.TUPLE_TAG) {
```

Comment:
Don't we still need the null check? If the constructor https://github.com/ballerina-platform/ballerina-lang/blob/2a8585fda1366cbc759ed5323909476aa0f9719d/bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValue.java#L195 is used, type could be null?

Reply:
Was running a build to see if anything was failing. With this constructor it probably could, will add the check